src/submodule.c

```diff
@@ -1854,6 +1854,17 @@ static void submodule_release(git_submodule *sm)
 	git__free(sm);
 }
 
+int git_submodule_dup(git_submodule **out, git_submodule *source)
+{
+	GIT_ASSERT_ARG(out);
+
+	if (source != NULL)
+		GIT_REFCOUNT_INC(source);
+
```

Comment:
Is there a reason `source` is optional?  I would make it required.

```suggestion
	GIT_ASSERT_ARG(source);

	GIT_REFCOUNT_INC(source);

```

Reply:
@ethomson 
Fixed